cmd/state-transition-update/main.go

```diff
@@ -87,6 +87,10 @@ func run() (rerr error) {
 		logging.Error("Could not determine installation path to check for installation: %s", errs.JoinMessage(err))
 	} else {
 		newStatePath := appinfo.StateApp(newInstallPath).Exec()
+
+		// ensure that path to transitional state tool is stored in the config file
+		transitionalStatePath := appinfo.StateApp().Exec()
+		cfg.Set(installation.CfgTransitionalStateToolPath, transitionalStatePath)
```

Comment:
This is a non-option. It's already out there. We need to assert it's transitional using the data that exists at the current time.

Reply:
I don't quite understand.  Note that the same block is also on line 165, where the transitional State Tool notifies us about its location, when it is running as `state _prepare`, so when it is being installed.

This block is just a fallback, in case the configuration got deleted inbetween, or the user has already done the v28->v29 transition before...

Reply:
Of course, it still leaves room for cases where we do not know about the transitional state tool location, but it would be very difficult to always find it.  We probably do not want to search the entire hard-drive...

Reply:
But it would only do this for people who update after we land this. People who have already updated are out of luck. Seeing as we've had VERY few updates so far and we're aiming to get this out today I guess we can work with that though.. as I don't know how else we would assert this either.

Reply:
Yes, exactly.  With this block, if they still have the PATH to the old (transitional) tool set, there is a high likelihood that they run this code block at least once.  It should even work for `state clean uninstall` actually.

If it is not on their PATH anymore, the stray executable will probably not bother them so much...